### PR TITLE
[IMP] ir.qweb: format monetary use field monetary method

### PR DESCRIPTION
**Pull request by Gorash**

### CODE REVIEW

**Comment by odony** [comment]
Let's discuss this in light of PR #11864. I think @antonylesuisse is slowly getting around the idea of keeping `t-esc-options`, and I think it would be fine to unify the rendering of monetary fields between t-field and t-esc-options (including the ``<span>``) in master (not 8.0 though).
Unifying both would simplify this PR.

PS: it would be great to add small descriptions to your PRs, and more importantly to improve your commit messages to explain **why** you're doing a change, not **what** your patches do (which we can see for ourselves in the diff) :wink:

**Comment by Gorash** [comment]
Ready to merge.
@odony Can you check it ?

File: openerp/addons/base/ir/ir_qweb/fields.py [hunk 1]
```diff
@@ -324,26 +315,39 @@ def record_to_html(self, record, field_name, options, values=None):
...
+    def record_to_html(self, record, field_name, options, values=None):
+        options = dict(options)
+        # for backward compatibility to remove after v10
+        if isinstance(options.get('display_currency'), str):
+            _logger.warning("Use new syntax for '%s' monetary widget 'display_currency' option (instead of deprecated JSON syntax) to improve speed." % options['display_currency'])
+            options['display_currency'] = safe_eval.safe_eval(options['display_currency'], mode='eval', globals_dict=values)
+        if isinstance(options.get('from_currency'), str):
+            _logger.warning("Use new syntax for '%s' monetary widget 'from_currency' option (instead of deprecated JSON syntax) to improve speed." % options['from_currency'])
+            options['from_currency'] = safe_eval.safe_eval(options['from_currency'], mode='eval', globals_dict=values)
+        # end backward
```
Comment (odony) [inline openerp/addons/base/ir/ir_qweb/fields.py]: move to value_to_html for compatibility with existing reports using t-esc-options

File: openerp/addons/base/ir/ir_qweb/ir_qweb.py [hunk 2]
```diff
@@ -65,10 +65,7 @@ def _get_template_cache_keys(self):
         return ['lang', 'inherit_branding', 'editable', 'translatable', 'edit_translations']
 
     # apply ormcache_context decorator unless in dev mode...
-    @tools.conditional(
-        'xml' not in tools.config['dev_mode'],
-        tools.ormcache('id_or_xml_id', 'tuple(map(options.get, self._get_template_cache_keys()))'),
-    )
+    @tools.ormcache('id_or_xml_id', 'tuple(map(options.get, self._get_template_cache_keys()))')
```
Comment (odony) [inline openerp/addons/base/ir/ir_qweb/ir_qweb.py]: todo: test in dev mode

File: openerp/addons/base/ir/ir_qweb/ir_qweb.py [hunk 3]
```diff
@@ -148,7 +145,7 @@ def _compile_directive_call_assets(self, el, options):
 
     @tools.conditional(
```
Comment (odony) [inline openerp/addons/base/ir/ir_qweb/ir_qweb.py]: add a one-line comment to explain that in non-xml-debug mode we want assets to be cached forever, and the admin can force a cache clear by restarting the server after updating the source code (or using the "Clear server cache" in debug tools - to be added later :-))

File: openerp/tools/cache.py [hunk 4]
```diff
@@ -6,6 +6,7 @@
 from collections import defaultdict
 from decorator import decorator
 from inspect import formatargspec, getargspec
+from time import time
```
Comment (odony) [inline openerp/tools/cache.py]: can be removed now

File: openerp/addons/base/ir/ir_qweb/fields.py (hunk 1 above)
Comment (Gorash) [inline openerp/addons/base/ir/ir_qweb/fields.py]: Oups, I forget that 'value_to_html' does not have 'values' as attributes, I can't move this part :-/

**Pull request merged**